processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -390,50 +366,18 @@ func (c *WatchClient) extractNamespaceAttributes(namespace *api_v1.Namespace) ma
 	tags := map[string]string{}
 
 	for _, r := range c.Rules.Labels {
-		if r.From == MetadataFromNamespace {
-			if r.KeyRegex != nil {
-				for k, v := range namespace.Labels {
-					if r.KeyRegex.MatchString(k) && v != "" {
-						name := fmt.Sprintf("k8s.namespace.labels.%s", k)
-						tags[name] = v
-					}
-				}
-			} else if v, ok := namespace.Labels[r.Key]; ok {
-				tags[r.Name] = c.extractField(v, r)
-			}
-		}
+		r.extractFromNamespaceMetadata(namespace.Labels, tags, "k8s.namespace.labels.%s")
 	}
 
 	for _, r := range c.Rules.Annotations {
-		if r.From == MetadataFromNamespace {
-			if r.KeyRegex != nil {
-				for k, v := range namespace.Annotations {
-					if r.KeyRegex.MatchString(k) && v != "" {
-						name := fmt.Sprintf("k8s.namespace.annotations.%s", k)
-						tags[name] = v
-					}
-				}
-			} else if v, ok := namespace.Annotations[r.Key]; ok {
-				tags[r.Name] = c.extractField(v, r)
-			}
-		}
+		r.extractFromNamespaceMetadata(namespace.Annotations, tags, "k8s.namespace.annotations.%s")
 	}
 
 	return tags
 }
 
 func (c *WatchClient) extractField(v string, r FieldExtractionRule) string {
```

Comment:
Is this function still being used?

Reply:
ah it's only used in UT now, the UT can be moved from WatchClient to FieldExtractionRule.

Reply:
updated.